### PR TITLE
Introduce x-aws-autoscale to support CPU based autoscaling

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
Introduced `x-aws-autoscaling` and configure autoscaling policy with a CPU percent target

**Related issue**
https://github.com/docker/compose-cli/issues/668

/test-ecs

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**
image

### CODE REVIEW

File: ecs/autoscaling_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+services:
+  foo:
+    image: hello_world
+    deploy:
+      x-aws-autoscaling: 75
+`)
+	target := template.Resources["FooScalableTarget"].(*autoscaling.ScalableTarget)
+	assert.Check(t, target != nil)
+	policy := template.Resources["FooScalingPolicy"].(*autoscaling.ScalingPolicy)
+	if policy == nil || policy.TargetTrackingScalingPolicyConfiguration == nil {
```
Comment (gtardif) [inline ecs/autoscaling_test.go:37]: nit: why `if` and not `assert.Check()` or `assert.Assert()` here ?

**Review by gtardif — APPROVED** [review]

File: ecs/autoscaling_test.go (hunk 1 above)
Comment (ndeloof) [reply ecs/autoscaling_test.go:37]: I first used `assert.Check(t, policy != nil)` but then linter complain about possible nil pointer

Comment (ndeloof) [reply ecs/autoscaling_test.go:37]:
```golang
	assert.Check(t, policy == nil)
	assert.Check(t, policy.TargetTrackingScalingPolicyConfiguration == nil)
	assert.Check(t, policy.TargetTrackingScalingPolicyConfiguration.TargetValue == float64(75))
```

```
#15 0.364 golangci-lint run --timeout 10m0s ./...
#15 4.729 ecs/autoscaling_test.go:38:25: SA5011: possible nil pointer dereference (staticcheck)
#15 4.729       assert.Check(t, policy.TargetTrackingScalingPolicyConfiguration == nil)
#15 4.729                              ^
#15 4.729 ecs/autoscaling_test.go:39:25: SA5011: possible nil pointer dereference (staticcheck)
#15 4.729       assert.Check(t, policy.TargetTrackingScalingPolicyConfiguration.TargetValue == float64(75))
```

Did I miss something ?

**Review by aiordache — APPROVED** [review]
LGTM

**Pull request merged**